src/sockaddr.rs

```diff
@@ -30,20 +30,38 @@ use std::net;
 
 use crate::bindings;
 
-pub fn to_sockaddr(addr: net::Ipv4Addr) -> bindings::sockaddr {
-    let mut sockaddr_in: nix::libc::sockaddr_in = unsafe { std::mem::zeroed() };
+pub use bindings::sockaddr as sockaddr;
 
-    sockaddr_in.sin_family = nix::libc::AF_INET as u16;
-    sockaddr_in.sin_addr = nix::libc::in_addr {
-        s_addr: u32::from_le_bytes(addr.octets()),
-    };
-    sockaddr_in.sin_port = 0;
-
-    unsafe { std::mem::transmute(sockaddr_in) }
+impl From<net::Ipv4Addr> for bindings::sockaddr {
+    fn from(addr: net::Ipv4Addr) -> Self {
+        let mut sockaddr_in: nix::libc::sockaddr_in = unsafe { std::mem::zeroed() };
+    
+        sockaddr_in.sin_family = nix::libc::AF_INET as u16;
+        sockaddr_in.sin_addr = nix::libc::in_addr {
+            s_addr: u32::from_le_bytes(addr.octets()),
+        };
+        sockaddr_in.sin_port = 0;
+    
+        unsafe { std::mem::transmute(sockaddr_in) }
+    }
 }
+impl From<[u8; 6]> for bindings::sockaddr {
+    fn from(addr: [u8; 6]) -> Self {
+        let mut addr = addr.to_vec();
+        addr.append(&mut [0x00; 8].to_vec());
+        
+        bindings::sockaddr { sa_family: nix::libc::ARPHRD_ETHER, sa_data: addr.iter().map(|x| *x as i8).collect::<Vec<i8>>().try_into().unwrap() }
+    }
+}
```

Comment:
We should avoid allocations if we can. The code snippet below is the implementation without allocating:
```rust
    let addr = [0u8; 8];
    let mut padded_addr = [0u8; 14];
    
    padded_addr[..8].copy_from_slice(&addr);
    
    let i8_addr = padded_addr.map(|v| v as i8);
```